services/base.spec.js

```diff
@@ -24,6 +24,29 @@ class DummyService extends BaseService {
 }
 
 describe('BaseService', () => {
+  describe('_makeBadgeData', function () {
```

Comment:
We have all cases with `text` tested here and that's great! Would you like to add tests for other properties? There's a lot of code connected to colors which isn't tested here.

Reply:
Yea, I'd love to add more tests, though maybe we can merge this and tag-team the tests. There is plenty of untested functionality here.